modules/ppcp-agentic-commerce/services.php

```diff
@@ -145,27 +149,57 @@
 	},
 
 	// Validation services.
-	'agentic.validation.processor'        => static function ( ContainerInterface $c ): CartValidationProcessor {
+	'agentic.validation.processor'                 => static function ( ContainerInterface $c ): CartValidationProcessor {
 		return new CartValidationProcessor(
 			$c->get( 'agentic.logger' )
 		);
 	},
-	'agentic.validator.product'           => static function ( ContainerInterface $c ): ProductValidator {
+	'agentic.validator.product'                    => static function ( ContainerInterface $c ): ProductValidator {
 		return new ProductValidator(
 			$c->get( 'agentic.helper.product-manager' )
 		);
 	},
-	'agentic.validator.inventory'         => static function ( ContainerInterface $c ): InventoryValidator {
+	'agentic.validator.inventory'                  => static function ( ContainerInterface $c ): InventoryValidator {
 		return new InventoryValidator(
 			$c->get( 'agentic.helper.product-manager' )
 		);
 	},
+	'agentic.validator.coupon.discount-calculator' => static function ( ContainerInterface $c ): DiscountCalculator {
+		return new DiscountCalculator(
+			$c->get( 'agentic.helper.product-manager' )
+		);
+	},
+	'agentic.validator.coupon.context-builder'     => static function ( ContainerInterface $c ): ContextBuilder {
+		return new ContextBuilder(
+			$c->get( 'agentic.helper.product-manager' ),
+			$c->get( 'agentic.validator.coupon.discount-calculator' )
+		);
+	},
+	'agentic.validator.coupon.resolution-builder'  => static function (): ResolutionBuilder {
+		return new ResolutionBuilder();
+	},
+	'agentic.validator.coupon'                     => static function ( ContainerInterface $c ): CouponValidator {
+		return new CouponValidator(
+			$c->get( 'agentic.validator.coupon.context-builder' ),
+			$c->get( 'agentic.validator.coupon.discount-calculator' ),
+			$c->get( 'agentic.validator.coupon.resolution-builder' )
+		);
+	},
 
-	// REST endpoints.
-	'agentic.response.factory'            => static function (): ResponseFactory {
-		return new ResponseFactory();
+	// Response services.
+	'agentic.response.applied-coupons-builder'     => static function ( ContainerInterface $c ): \WooCommerce\PayPalCommerce\AgenticCommerce\CartValidation\CouponValidator\AppliedCouponsBuilder {
```

Comment:
Suggestion: Use the project standard of importing classes at the top, and not using FQCN in the module code